src/bootstrap/src/lib.rs

```diff
@@ -579,6 +585,45 @@ impl Build {
         }
     }
 
+    /// Updates a submodule, and exits with a failure if submodule management
+    /// is disabled and the submodule does not exist.
+    ///
+    /// The given `err_hint` will be shown to the user if the submodule is not
+    /// checked out.
+    pub fn require_and_update_submodule(&self, submodule: &str, err_hint: Option<&str>) {
+        // When testing bootstrap itself, it is much faster to ignore
+        // submodules. Almost all Steps work fine without their submodules.
+        if cfg!(test) && !self.config.submodules() {
+            return;
+        }
+        let relative_path = Path::new(submodule);
+        self.update_submodule(relative_path);
+        let absolute_path = self.config.src.join(relative_path);
+        if dir_is_empty(&absolute_path) {
+            let maybe_enable = if !self.config.submodules()
+                && self.config.rust_info.is_managed_git_subrepository()
+            {
+                "\nConsider setting `build.submodules = true` or manually initializing the submodules."
+            } else {
+                ""
+            };
+            let err_hint = err_hint.map_or_else(String::new, |e| format!("\n{e}"));
```

Comment:
Sorry, just to clarify here. There is already a newline being prepended (so that it is on a line by itself). Are you saying there should be a second newline so there is a blank line separating them?

Reply:
For context, the error currently looks like this:

```
submodule library/backtrace does not appear to be checked out, but it is required for this step
Consider setting `build.submodules = true` or manually initializing the submodules.
The submodule is required for the standard library and the main Cargo workspace.
Build completed unsuccessfully in 0:00:00
```

Reply:
Sorry, I somehow read `map_or_else` as `unwrap_or_else` :man_facepalming: Nevermind.